### PR TITLE
Waku_Subscribe request serialization and test

### PR DESCRIPTION
**Pull request by TalhaAli00**
Added waku_subscribe request object with test

### CODE REVIEW

**Review by ligi — CHANGES_REQUESTED** [review]
some minor things I saw when looking over the code - maybe we should also make Expiry and TTL value classes - noticed when seeing the adapter tests

File: walletconnectV2/src/test/kotlin/org/walletconnect/walletconnectv2/common/network/adapters/ExpiryAdapterTest.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+internal class ExpiryAdapterTest {
+    private val moshi = Moshi.Builder()
+        .add(ExpiryAdapter)
+        .add(KotlinJsonAdapterFactory())
+        .build()
+
+    @Test
+    fun fromJson() {
+    }
```
[the remark this answers is not in the document]
Comment (TalhaAli00) [reply walletconnectV2/src/test/kotlin/org/walletconnect/walletconnectv2/common/network/adapters/ExpiryAdapterTest.kt:19]: I've removed from current branch. Updates will be in next PR

**Comment by TalhaAli00** [comment]
> some minor things I saw when looking over the code - maybe we should also make Expiry and TTL value classes - noticed when seeing the adapter tests

The started off with value classes but there was an issue deserializing the classes. I have left the objects as data classes for now until I get a chance to dig deeper

File: walletconnectV2/src/test/kotlin/org/walletconnect/walletconnectv2/common/network/adapters/ExpiryAdapterTest.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    @Test
+    fun toJson() {
+        val expiry = Expiry(Random.nextLong())
```
[the remark this answers is not in the document]
Comment (TalhaAli00) [reply walletconnectV2/src/test/kotlin/org/walletconnect/walletconnectv2/common/network/adapters/ExpiryAdapterTest.kt:23]: I understand your point and I have updated the code to no longer use Random

**Pull request merged**